File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>


bool is_armstrong_number(int candidate)
{
    int copy = candidate;
    int num_digits = 0, sum_powers = 0;
    while(copy)
    {
        copy /= 10;
        num_digits++;
    }
    copy = candidate;
    while(copy)
    {
        int current = copy % 10;
        sum_powers += (int)pow(current, num_digits);
        copy /= 10;
    }
    return sum_powers == candidate;
}

File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    static int pow(int base, int exponent) {
        int result = 1;
        for (int count = 0; count < exponent; count++) {
            result *= base;
        }
        return result;
    }
    boolean isArmstrongNumber(int numToCheck) {
        int numDigits = 0;
        int copy = numToCheck;
        
        while (copy > 0) {
            numDigits++;
            copy /= 10;
        }

        copy = numToCheck;
        int sum = 0;
        while (copy > 0) {
            sum += (pow(copy % 10, numDigits));
            copy /= 10;
        }

        return sum == numToCheck;
    }

}
